### ISSUE TITLE
.gitconfig not copied to remote container

### ISSUE THREAD

**Issue by slaytanic** [microsoft/vscode-remote-release#6124]
I'm using devcontainers with Docker on Windows, according to the documentation at https://code.visualstudio.com/docs/remote/containers#_sharing-git-credentials-with-your-container it says `.gitconfig` is copied from the host, this does not seem to be the case when using Git for Windows at least which stores `.gitconfig` in `C:\Users\<username>`

- VSCode Version: 1.63.2
- Local OS Version: Windows 10.0.19044
- Remote OS Version: mcr.microsoft.com/vscode/devcontainers/ruby:0-3
- Remote Extension/Connection Type: Docker

Steps to Reproduce:

1. Run inside a devcontainer
2. .gitconfig is missing

Does this issue occur when you try this locally?: Yes
Does this issue occur when you try this locally and all extensions are disabled?: Yes

**Comment by chrmarti** [comment]
Could you rebuild the container and append the log from that? (`F1` > `Remote-Containers: Show Container Log`)

**Comment by slaytanic** [comment]
Sure, here it is
[devcontainer.log](https://github.com/microsoft/vscode-remote-release/files/7807653/devcontainer.log)

**Comment by chrmarti** [comment]
Do you have `"remote.containers.copyGitConfig"` set to false either in the devcontainer.json or your user settings?

**Comment by slaytanic** [comment]
> Do you have `"remote.containers.copyGitConfig"` set to false either in the devcontainer.json or your user settings?

I don't, but I just noticed that on a different project it was working and at first I assumed it was because it was not using docker-compose, so I tried creating a new project with compose, and one without it and it works as expected on both.

[nocomposetest.log](https://github.com/microsoft/vscode-remote-release/files/7813669/nocomposetest.log)
[composetest.log](https://github.com/microsoft/vscode-remote-release/files/7813671/composetest.log)

I'm not sure what could be making it skip that step on that other project. If you've got any other pointers on things to check I'd appreciate it, if not I guess it can be chalked off to something of the environment.
